core/src/Kokkos_Parallel_Reduce.hpp

```diff
@@ -30,37 +30,54 @@ static_assert(false,
 
 namespace Kokkos {
 
-template <class Scalar, class Space>
-struct Sum {
- public:
-  // Required
-  using reducer    = Sum<Scalar, Space>;
-  using value_type = std::remove_cv_t<Scalar>;
-  static_assert(!std::is_pointer_v<value_type> && !std::is_array_v<value_type>);
+/** \brief Class offering functionalities common to all reducers
+ *
+ * In order to be a valid reducer, a class must implement the functions and
+ * define the types offered in this class.
+ * To facilitate implementation, a new reducer class can simply inherit from
+ * BaseReducer.
+ */
```

Comment:
We generally prefer C++-style comment
```suggestion
// \brief Class offering functionalities common to all reducers
//
// In order to be a valid reducer, a class must implement the functions and
// define the types offered in this class.
// To facilitate implementation, a new reducer class can simply inherit from
// BaseReducer.
```

Reply:
Is the coherence of style inside of a file an issue? Other comments in the file use C-style comments (line 1317 for instance), should they be changed to or should they only be changed when modified?

Reply:
It depends.  I would fix when we touch that code.